### PR TITLE
feat(migrate): don't create local `view` folder if no views are introspected.

### LINKED ISSUE #18611: views: don't create local `views` folder on introspection if no views are found

### PR DESCRIPTION
**Pull request by jkomyno**
Closes https://github.com/prisma/prisma/issues/18611.

### CODE REVIEW

File: packages/migrate/src/utils/setupPostgres.ts [hunk 1]
```diff
@@ -53,3 +53,21 @@ export async function tearDownPostgres(options: SetupParams) {
...
+  const { connectionString } = options
+
+  // Connect to default db
+  const dbDefault = new Client({
+    connectionString,
+  })
+  await dbDefault.connect()
+  await dbDefault.query(query)
+  await dbDefault.end()
+}
```
Comment (jkomyno) [inline packages/migrate/src/utils/setupPostgres.ts:73]: In the future, we may want to have similar functions such as this, but for other providers.

File: packages/internals/src/utils/fs-functional.ts [hunk 2]
```diff
@@ -28,7 +28,11 @@ export const writeFile = ({ path, content }: { path: string; content: string })
 
 export const removeDir = (dir: string) =>
   pipe(
-    TE.tryCatch(() => fs.promises.rmdir(dir, { recursive: true }), createTaggedSystemError('fs-remove-dir', { dir })),
+    TE.tryCatch(
+      // @ts-ignore: `rm` exists since Node.js 14.14.0, but doesn't seem to appear in all @types/node packages used by Prisma
```
[the remark this answers is not in the document]
Comment (Jolg42) [reply packages/internals/src/utils/fs-functional.ts:32]: I could not resist updating the version to latest everywhere 😄 
https://github.com/prisma/prisma/pull/18904

[the remark this answers is not in the document]
Comment (jkomyno) [reply packages/internals/src/utils/fs-functional.ts]: Note: my other PR (https://github.com/prisma/prisma/pull/18899) fixed this problem. I could in fact get rid of `@ts-ignore`.

**Review by Jolg42 — APPROVED** [review]
Should the title of this PR actually be "delete local view folder if no views are introspected"?

**Review by Jolg42 — CHANGES_REQUESTED** [review]
Should the title of this PR actually be "delete local view folder if no views are introspected"?

(sorry meant to add as a comment)

**Review by Jolg42 — CHANGES_REQUESTED** [review]
This is scary, but it's the current general behavior about the `views`directory from what I tested:

When the Prisma schema is at `prisma/schema.prisma` then the `prisma/views` directory and all its contents are deleted.
When the Prisma schema is at `schema.prisma` then the `views` directory and all its contents are deleted.

See the commit I just did with an extra test [`d4a5fb4` (#18848)](https://github.com/prisma/prisma/pull/18848/commits/d4a5fb42821a83cd5211d5e0b3d9616e486df377)

I don’t think we can keep this behavior, it’s dangerous, and I would not trust Prisma CLI if as a user I discover things like this.
Mentioning it in docs feels wrong, and personally I would be very surprised to find that only in docs.
Silently deleting is also weird but deleting with a log does not help so much either because it's after the fact. 🤔  

Deleting a `views` directory outside a `prisma` directory feels the most wrong part here to me. And this PR amplifies the problem by deleting it if no views are introspected.

**Comment by jkomyno** [comment]
This PR is deprecated in favor of https://github.com/prisma/prisma/pull/19131

**Pull request closed without merging**